packages/devtools_app/lib/src/inspector/inspector_screen.dart

```diff
@@ -70,7 +72,7 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
 
   LinkedHashMap<String, DevToolsError> _errors =
       LinkedHashMap<String, DevToolsError>();
-  int _selectedErrorIndex;
+  final ValueNotifier<int> _selectedErrorIndex = ValueNotifier<int>(null);
```

Comment:
let's move this to inspector controller and expose a public ValueListenable getter that returns `_selectedErrorIndex`. You can expose a public method `selectError(int index)` that performs what is in `_onSelectedErrorIndexChanged` below.

Reply:
I don't know if I've understood correctly, but I think I've done this in eee1fd77. I ended up moving quite a bit of code into there, and ErrorNavigator became stateless again. Let me know if I've misunderstood/gone too far!

I did wonder if we should move the error stuff into another class that hangs off InspectorController too, as it's quite large and this seems to have made it larger!